src/shiftentercommand.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @module enter/shiftentercommand
+ */
+
+import Command from '@ckeditor/ckeditor5-core/src/command';
+
+/**
+ * ShiftEnter command. It is used by the {@link module:enter/shiftenter~ShiftEnter ShiftEnter feature} to handle
+ * the <kbd>Shift</kbd>+<kbd>Enter</kbd> keystroke.
+ *
+ * @extends module:core/command~Command
+ */
+export default class ShiftEnterCommand extends Command {
+	/**
+	 * @inheritDoc
+	 */
+	execute() {
+		const model = this.editor.model;
+		const doc = model.document;
+
+		model.change( writer => {
+			softBreakAction( model, writer, doc.selection );
+			this.fire( 'afterExecute', { writer } );
+		} );
+	}
+
+	refresh() {
+		const model = this.editor.model;
+		const doc = model.document;
+
+		this.isEnabled = isEnabled( model.schema, doc.selection );
+	}
+}
+
+// Checks whether the shiftEnter command should be enabled in the specified selection.
+//
+// @param {module:engine/model/schema~Schema} schema
+// @param {module:engine/model/selection~Selection|module:engine/model/documentselection~DocumentSelection} selection
+function isEnabled( schema, selection ) {
+	if ( selection.rangeCount > 1 ) {
```

Comment:
First of all – there are no tests of `ShiftEnterCommand#isEnabled`. There should be. Those tests should explain the logic in this function.

The other things is – why is this command disabled on multiple ranges? I guess you did that on purpose to simplify it and it makes sense to do that, but that requires a comment in the code (because, in the future, we'll need to change that).